processor/src/test/java/org/mapstruct/ap/test/collection/map/AnnotatedMapMappingTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.collection.map;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mapstruct.ap.test.collection.map.other.ImportedType;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.GeneratedSource;
+
+import static java.lang.System.lineSeparator;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+
+/**
+ * Test for implementation of {@code Map} mapping methods with @{@link org.mapstruct.Mapping Mappings}
+ *
+ * @author Oliver Erhart
+ */
+@WithClasses({
+    AnnotatedSourceTargetMapper.class,
+    CustomNumberMapper.class,
+    Source.class,
+    Target.class,
+    ImportedType.class
+})
+@IssueKey("3303")
+public class AnnotatedMapMappingTest {
+
+    @RegisterExtension
+    final GeneratedSource generatedSource = new GeneratedSource();
+
+    @ProcessorTest
+    public void shouldContainTwoKeyMappingMapsInConstructor() {
```

Comment:
I would prefer if we have a fixture for this and we test the contents. Makes it a bit easier to see what gets generated as well